### ISSUE TITLE
setViewportSize doesn't work properly on some pages, modals etc.

### ISSUE THREAD

**Issue by vladavr3** [microsoft/playwright#8039]
### Playwright version

1.13.0 Java

### Operating system

Windows

### What browsers are you seeing the problem on?

Chromium, Firefox, WebKit

### What happened? / Describe the bug

I'm running tests on the 23inch monitor with the display resolution set to 1920x1080. When I set setViewportSize to 1920x1080, I'm missing the part of the modal because the opened page is too big and it cannot be scrolled down. I've tried to maximize the window manually while tests are running, but the page doesn't scale down to my device's viewport size.
I'm attaching the screenshots, pay attention to the footer and the buttons, and to the scroll box.
Actual:
![actual](https://user-images.githubusercontent.com/40892855/128513572-edf6ac11-dc0d-4d28-8ccb-8f25cedaa826.PNG)

Expected
![expected](https://user-images.githubusercontent.com/40892855/128513589-0215667a-92dd-4536-a32a-e469bd534480.png)

This can be also checked on https://www.orangehrm.com/orangehrm-30-day-trial/ where part of the footer is missing when run with the settings described 

### Code snippet to reproduce your bug

```shell
The only settings that I use:
Browser.NewContextOptions contextOptions = new Browser.NewContextOptions();
        contextOptions.setViewportSize(1920,1080);

BrowserType.LaunchOptions launchOptions = new BrowserType.LaunchOptions()
                .setHeadless(false)
```

**Comment by mxschmitt** [comment]
I think for https://www.orangehrm.com/orangehrm-30-day-trial/ its expected, I also have a 1920x1080 monitor and I can't see the footer in any of the browsers (also without Playwright). If you use Playwright for end-to-end testing, click etc. should automatically scroll to the correct position when e.g. clicking on a button which is not on the viewport.

For the first two pictures, we would need to have a URL so we can reproduce it ourselves.

**Comment by vladavr3** [comment]
I'm not sure why we use the headful mode then? :)
You can scroll down to see the complete footer without Playwright, but you are not able to see it in tests which Playwright runs because you can't scroll down, so I'm not sure how is this expected. I would expect that everything I see on the page without Playwright to be able to see when the tests start, otherwise we can run headless all the time.
The thing is that Playwright cannot scroll to the bottom to be able to see the footer, the window that is opened is too large.

Unfortunately, I cannot provide the URL as the application is still being developed, that's why I provided the test page link.

**Comment by mxschmitt** [comment]
Headful is usually used to debug tests. You can also scroll down with Playwright, see the following example which will display the whole footer:

```js
// @ts-check
const playwright = require('playwright');

(async () => {
  // Try to add 'firefox' to the list ↓
  for (const browserType of ['chromium']) {
    /** @type {import('playwright').Browser} */
    const browser = await playwright[browserType].launch();
    const context = await browser.newContext();
    const page = await context.newPage();
    await page.goto('https://www.orangehrm.com/orangehrm-30-day-trial/');
    await page.evaluate(() => window.scrollTo(0, document.body.scrollHeight))
    await page.screenshot({ path: `example-${browserType}.png` });
    await browser.close();
  }
})();
```

Playwright behaves like normal browsers behave. They have a viewport and optionally some parts which exceed the viewport, in your case the footer. You can manually scroll down, via JavaScript or you leverage the normal Playwright methods e.g. click which will automatically scroll down for you.

So from my side it seems like expected behaviour.

**Comment by vladavr3** [comment]
I probably missed the section, I'm using Java, sorry about that.
Anyways, have you tried to set setVewportSize to 1920x1080? In this case I set:
` page.keyboard().press("Control");
  page.keyboard().down("End");`
Before that, I focus on any element by click on it.
What I get with Playwright view is this image and cannot scroll down anymore:
![actual_orange](https://user-images.githubusercontent.com/40892855/128526083-821d12fc-c10c-4832-a5ed-dfe6dda027fe.PNG)

And this is what I see in the browser:
![expected_orange](https://user-images.githubusercontent.com/40892855/128526132-eb011f75-73ba-401a-a4b3-1f320f6a4741.png)

As you can see almost half of the footer is missing. I've tried with setChannel also but to no avail.
Basically, I cannot debug anything that is not in my view, so I'm still not sure how is this expected.

**Comment by mxschmitt** [comment]
Java uses the same foundation as the main Playwright project. You can also execute `Page.Evaluate("() => window.scrollTo(0, document.body.scrollHeight)")` which is a better way of scrolling down.

btw. what is the scenario you are trying to solve?

**Comment by vladavr3** [comment]
The scenario is the basic registration form from the initial post. Click on the button was a bit tricky, it wouldn't work without page.waitForTimeout(1000) and I wasn't sure why. Basically, I cannot see what's going on with the button because I don't see it with current viewport settings.

**Comment by pavelfeldman** [comment]
I'm not sure we'll be able to do anything with this one - your screen resolution can't fit your window size and hence the debugging story is compromised. I understand that this is frustrating, but I'm afraid you need bump the resolution for window to fit...

**Comment by xmmw** [comment]
i also have this problem, did you fix it? the browser does not show the whole footer
